connectors/sources/mysql.py

```diff
@@ -111,31 +117,74 @@ async def ping(self):
             raise
 
     async def _stream_rows(self, database, table, query):
-        size = int(self.configuration.get("fetch_size", DEFAULT_FETCH_SIZE))
-        logger.debug(f"Streaming {database}.{table} {size} rows at a time")
+        """Executes the passed query on the MySQL server.
 
-        async with self.connection_pool.acquire() as connection:
-            count = 0
-            async with connection.cursor(aiomysql.cursors.SSCursor) as cursor:
-                await cursor.execute(query)
+        Args:
+            database (str): Name of database
+            table (str): Name of table
+            query (str): MySql query to be executed.
 
-                # sending back column names
-                yield [column[0] for column in cursor.description]
+        Yields:
+            list: Column names and query response
+        """
 
-                while True:
-                    rows = await cursor.fetchmany(size=size)
+        size = int(self.configuration.get("fetch_size", DEFAULT_FETCH_SIZE))
+        logger.debug(f"Streaming {database}.{table} {size} rows at a time")
 
-                    if len(rows) == 0:
+        # retry: Current retry counter
+        # yield_once: Yield(fields) once flag
+        retry = yield_once = 1
+
+        # rows_fetched: Rows fetched counter
+        # cursor_position: Mocked cursor position
+        rows_fetched = cursor_position = 0
+        while retry <= int(self.configuration.get("retry_count", DEFAULT_RETRY_COUNT)):
```

Comment:
you can store `retry_count` on the class to avoid doing this `int(self....)` on every loop